test/relations.test.js

```diff
@@ -1849,6 +2500,28 @@ describe('relations', function () {
       });
     });
 
+    it('can be used to query data with promises', function (done) {
```

Comment:
We need to promisify datasource methods too, but that's out of scope of this PR.